src/libYARP_OS/src/RFModuleFactory.cpp

```diff
@@ -0,0 +1,188 @@
+/*
+* Copyright (C) 2006-2018 Istituto Italiano di Tecnologia (IIT)
+* All rights reserved.
+*
+* This software may be modified and distributed under the terms of the
+* BSD-3-Clause license. See the accompanying LICENSE file for details.
+*/
+#include <yarp/os/YarpPluginSelector.h>
+#include <yarp/os/YarpPluginSettings.h>
+#include <yarp/os/YarpPlugin.h>
+#include <yarp/os/RFModuleFactory.h>
+#include <map>
+using namespace std;
+using namespace yarp::os;
+
+class RFModuleSelector : public YarpPluginSelector
+{
+    virtual bool select(Searchable& options) override
+    {
+        return options.check("type",Value("none")).asString() == "RFModule";
+    }
+};
+
+struct SharedRFPlugin
+{
+    YarpPlugin<RFModule>         yarpPlugin;
+    SharedLibraryClass<RFModule> sharedLibClass;
+    RFModuleSelector             selector;
+};
+
+struct RFPlugin::RFPlugin_Private
+{
+    RFPlugin_Private() = default;
+
+    string          alias;
+    string          name;
+    string          command;
+    int             threadID{0};
+    SharedRFPlugin* shared{nullptr};
+    
+    RFModule*       module{nullptr};
+    ~RFPlugin_Private()
+    {
+        delete shared;
+    }
+};
+
+RFPlugin::RFPlugin() :
+    impl(new RFPlugin_Private)
+{
+}
+
+RFPlugin::~RFPlugin()
+{
+    delete impl;
+}
+
+string RFPlugin::getCmd()
+{
+    return impl->command;
+}
+
+string RFPlugin::getAlias()
+{
+    return impl->alias;
+}
+
+int RFPlugin::getThreadKey()
+{
+    return impl->module->getThreadKey();
+}
+
+void RFPlugin::close()
+{
+    impl->module->stopModule();
+}
+
+bool RFPlugin::isRunning()
+{
+    return !impl->module->isStopping();
+}
+
+bool RFPlugin::open(const string& inCommand)
+{
+    ResourceFinder     rf;
+    string name = inCommand.substr(0, inCommand.find(" "));
+
+    char* str = new char[inCommand.size() + 1];
+    memcpy(str, inCommand.c_str(), inCommand.size());
+    str[inCommand.size()] = '\0';
+
+    enum { kMaxArgs = 64 };
+    int argc = 0;
+    char* argv[kMaxArgs];
+
+    char* p2 = strtok(str, " ");
+    while (p2 && argc < kMaxArgs - 1)
+    {
+        argv[argc++] = p2;
+        p2 = strtok(0, " ");
+    }
+    argv[argc] = 0;
```

Comment:
@lrapetti this is probably the code that we need also in gazebo-yarp-plugins to load command line-like parameters from string. @aerydna it would be good to refactor it in a reusable function.

Reply:
sure! im finally back on this pr

Reply:
i've put it in a separate function.. but actually in the same file.. maybe things like that should be put in a dedicated library. something like libYARP_utils